### ISSUE TITLE
Strange behavior of loss function in WALSMatrixFactorization TF 1.8

### ISSUE THREAD

**Issue by NatLun091238** [tensorflow/tensorflow#21991]
Our current implementation of recommendation system use TF 1.8 and WALS algorithm. The model was trained using self.fit(input_fn=input_fn) and ML Engine with run time version 1.8. Data set was formed following [example](https://github.com/GoogleCloudPlatform/training-data-analyst/blob/master/courses/machine_learning/deepdive/10_recommend/wals.ipynb) using tensorflow.train.Example(...) Extraction from training logs shown below. 

[Extraction from training logs](https://drive.google.com/open?id=1HS4dPXC31Qv42YAWb_PT5y_C_uFD03GM) 

The fit was performed with some default parameters. The loss value did decreased on second evaluation. However loss did not changed after that. The final Root weighted squared error (rwse) in this training became 0.126.

Hyperparameter tuning was performed later and the best parameter set was used in the following training. The result of that training is shown below.

[Post hyperparameter tuning log extraction](https://drive.google.com/open?id=1zrt1jsfVdFAtuEegzquV-8sU8AJtlmKY)

Tree things to note here. First, the loss value at the beginning is lower than at later evaluation steps. Low value in the beginning most likely due to choice of parameters from the results of hyperparameter tuning. Increase of the loss value later on looks strange. Second, it’s unchanged loss value after second evaluation. This pattern remains the same while self.fit(input_fn=input_fn) is used for model training. Third, the final rwse in this training became 0.487 while during hyperparameter tuning with the same parameter set rwse=0.015. 

The question is if anyone observed something similar? Could the reason of unchanged loss value be due to fixed learning rate? If yes, is it possible to change it while using WALSMatrixFactorization class and self.fit(input_fn=input_fn)??

------------------------

### System information

- GCP VM instance  created-with-datalab-version	20180503

- TensorFlow version 1.8.0

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by NatLun091238** [comment]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes, but my code tightly follows [examlple](https://github.com/GoogleCloudPlatform/training-data-analyst/blob/master/courses/machine_learning/deepdive/10_recommend/wals.ipynb)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.4 LTS
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: N/A
- **TensorFlow version (use command below)**: TensorFlow version 1.8.0
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

**Comment by NatLun091238** [comment]
I used fully managed by Google n1-highmem-16 (16 vCPUs, 104 GB memory) instance for code development. Further packaged model was trained and later hyperparameter tuned with Google cloud ML engine.

**Comment by NatLun091238** [comment]
Yes, you have misinterpreted a bug. I compile a code which reproduce the bug, namely **loss value does not update during the training**, on both Windows 10 and Ubuntu 16.04, as well as on TF 1.8 and TF 1.10.0.

**Comment by NatLun091238** [comment]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
yes [the code to reproduce the bug](https://drive.google.com/open?id=1ahzI5vZQwqXB0TsjI8dSVCGGH7z-ut0f)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: both, Windows 10 and Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: A/N
- **TensorFlow installed from (source or binary)**: no, I used miniconda environment and pip install on Windows 10
- **TensorFlow version (use command below)**: both TensorFlow version 1.8.0 and TensorFlow version 1.8.0
- **Python version**: both 2.7 and 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: [the code](https://drive.google.com/open?id=1ahzI5vZQwqXB0TsjI8dSVCGGH7z-ut0f)

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

> Line 1: 	Line 24: 2018-09-11 14:16:17,930 - tensorflow - INFO - loss = 1258.4746, step = 1
	Line 2: 	Line 25: 2018-09-11 14:16:17,930 - tensorflow - INFO - loss = 1258.4746, step = 1
	Line 3: 	Line 232: 2018-09-11 14:16:22,078 - tensorflow - INFO - loss = 42.56178, step = 101 (4.148 sec)
	Line 4: 	Line 233: 2018-09-11 14:16:22,078 - tensorflow - INFO - loss = 42.56178, step = 101 (4.148 sec)
	Line 5: 	Line 436: 2018-09-11 14:16:25,565 - tensorflow - INFO - loss = 42.56178, step = 201 (3.487 sec)
	Line 6: 	Line 437: 2018-09-11 14:16:25,565 - tensorflow - INFO - loss = 42.56178, step = 201 (3.487 sec)
	Line 7: 	Line 640: 2018-09-11 14:16:29,490 - tensorflow - INFO - loss = 42.56178, step = 301 (3.925 sec)
	Line 8: 	Line 641: 2018-09-11 14:16:29,490 - tensorflow - INFO - loss = 42.56178, step = 301 (3.925 sec)
	Line 9: 	Line 844: 2018-09-11 14:16:33,250 - tensorflow - INFO - loss = 42.56178, step = 401 (3.760 sec)
	Line 10: 	Line 845: 2018-09-11 14:16:33,250 - tensorflow - INFO - loss = 42.56178, step = 401 (3.760 sec)
	Line 11: 	Line 1048: 2018-09-11 14:16:36,489 - tensorflow - INFO - loss = 42.56178, step = 501 (3.239 sec)
	Line 12: 	Line 1049: 2018-09-11 14:16:36,489 - tensorflow - INFO - loss = 42.56178, step = 501 (3.239 sec)
	Line 13: 	Line 1252: 2018-09-11 14:16:40,309 - tensorflow - INFO - loss = 42.56178, step = 601 (3.820 sec)
	Line 14: 	Line 1253: 2018-09-11 14:16:40,309 - tensorflow - INFO - loss = 42.56178, step = 601 (3.820 sec)
	Line 15: 	Line 1456: 2018-09-11 14:16:43,931 - tensorflow - INFO - loss = 42.56178, step = 701 (3.623 sec)
	Line 16: 	Line 1457: 2018-09-11 14:16:43,931 - tensorflow - INFO - loss = 42.56178, step = 701 (3.623 sec)
	Line 17: 	Line 1534: 2018-09-11 14:16:45,488 - tensorflow - INFO - loss = 42.56
	Line 18: 	Line 1535: 2018-09-11 14:16:45,488 - tensorflow - INFO - loss = 42.56

As can seen from the log file extraction contained lines "loss =", the loss value **does not get updated** after second evaluation. 

### Source code / logs
Complete log could be found [here](https://drive.google.com/open?id=1vy0GDKK4HMdAirLGewM2vUa_61MGfc_x)

**Comment by reedwm** [comment]
Sorry for closing this issue. @walidk, can you take a look?

**Comment by harshini-gadige** [comment]
@walidk Could you please look into this.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @walidk: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @walidk: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @walidk: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @walidk: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
